conda_recipe_manager/parser/recipe_parser_convert.py

```diff
@@ -196,6 +197,45 @@ def _upgrade_selectors_to_conditionals(self) -> None:
                 self._patch_and_log(patch)
                 self._new_recipe.remove_selector(selector_path)
 
+    def _upgrade_source_section(self, base_package_paths: list[str]) -> None:
+        """
+        Upgrades/converts the `source` section(s) of a recipe file.
+        :param base_package_paths: Set of base paths to process that could contain this section.
+        """
+        for base_path in base_package_paths:
+            source_path = RecipeParser.append_to_path(base_path, "/source")
+            if not self._new_recipe.contains_value(source_path):
+                continue
+
+            # The `source` field can contain a list of elements or a single element (not encapsulated in a list).
+            # This logic sets up a list to iterate through that will handle both cases.
+            source_data = self._new_recipe.get_value(source_path)
+            source_paths = []
+            if isinstance(source_data, list):
+                for x in range(len(source_data)):
+                    source_paths.append(RecipeParser.append_to_path(source_path, f"/{x}"))
+            else:
+                source_paths.append(source_path)
+
+            for src_path in source_paths:
+                # SVN and HG source options are no longer supported. This seems to have been deprecated a long
+                # time ago and there are unlikely any recipes that fall into this camp. Still, we should flag it.
+                if self._new_recipe.contains_value(RecipeParser.append_to_path(src_path, "svn_url")):
+                    self._msg_tbl.add_message(
+                        MessageCategory.WARNING, "SVN packages are no longer supported in the new format"
+                    )
+                if self._new_recipe.contains_value(RecipeParser.append_to_path(src_path, "hg_url")):
+                    self._msg_tbl.add_message(
+                        MessageCategory.WARNING, "HG (Mercury) packages are no longer supported in the new format"
```

Comment:
😢 My favorite system. Is there a more permanent name besides "the new format"?
```suggestion
                        MessageCategory.WARNING, "HG (Mercurial) packages are no longer supported in the new format"
```

Reply:
I have "V1" in a few places as that's what the `schema_version` field uses. The "old version" is then referred to as "V0".

But past that, I don't think there is an official nomenclature that anyone has come up with/is using.

Reply:
Also `Mercury` is the name rattler's docs used so I just copied the name from that.